spec/features/visitor/list_articles_spec.rb

```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+feature "List articles" do
+  let!(:articles) { create_list(:article, 2) }
+
+  background do
+    visit articles_path
+  end
+
+  scenario "Visitor lists articles" do
+    article1, article2 = articles
+
+    expect(page).to have_article(article1.decorate)
```

Comment:
I think feature spec should not user decorated methods to check check static texts on page, because if you will remove this decorator or move this logic in other place your spec will become broken. You can write this spec something like this:

```
let(:first_article) { create :article, title: "123", text: "456" }
let(:first_article_selector) { ".article-item[data-id='#{first_article.id}']" }

scenario "Visitor lists articles" do
    expect_to_have_first_article_data
    expect_to_have_second_article_data
  end

def expect_to_have_first_article_data
  within first_article_selector do
    expect(page).to have_content("123")
    expect(page).to have_content("456")
  end
end

def expect_to_have_second_article_data
...
end
```